root.go

```diff
@@ -45,13 +45,12 @@ type child struct {
 // the entry/link of the directory that pointed to the modified node).
 type parent interface {
 	// Method called by a child to its parent to signal to update the content
-	// pointed to in the entry by that child's name. The child sends as
-	// arguments its own information (under the `child` structure) and a flag
-	// (`fullsync`) indicating whether or not to propagate the update upwards:
-	// modifying a directory entry entails modifying its contents which means
-	// that its parent (the parent's parent) will also need to be updated (and
-	// so on).
-	updateChildEntry(c child, fullSync bool) error
+	// pointed to in the entry by that child's name. The child sends its own
+	// information in the `child` structure. As modifying a directory entry
+	// entails modifying its contents the parent will also call *its* parent's
+	// `updateChildEntry` to update the entry pointing to the new directory,
+	// this mechanism is in turn repeated until reaching the `Root`.
```

Comment:
We can probably just simplify this to:

> `updateChildEntry` notifies the parent that the child has been updated.

Reply:
> > notifies the parent that the child has been updated.

Not to someone who it not familiar with the way MFS works and what is implied in a content-addressed system (which is a large part of our contributor base), ideally this should be clearly documented in a separate high level document but we're far from that.